master/internal/api_trials.go

```diff
@@ -1525,6 +1456,26 @@ func (a *apiServer) ReportTrialMetrics(
 		experiment.AuthZProvider.Get().CanEditExperiment); err != nil {
 		return nil, err
 	}
+	defer func() {
+		// Notify searcher of validation metrics.
+		if metricGroup == model.ValidationMetricGroup {
```

Comment:
code style suggestion:

EITHER reduce indentation:

```go
if metricGroup != model.ValidationMetricGroup {
  return
}
```

OR don't defer pointless functions:

```go
if metricGroup == model.ValidationMetricGroup {
    defer func() {
    }()
}
```

take your pick.  I like the first because less indentation is good for readability, and I like the second because not calling a noop is also good for readability.

Reply:
chose the second, read better to me